qutebrowser/misc/sessions.py

```diff
@@ -428,8 +436,9 @@ def load(self, name, temp=False):
                     tab_to_focus = i
                 if new_tab.data.pinned:
                     tabbed_browser.set_tab_pinned(new_tab, new_tab.data.pinned)
+
             if tab_to_focus is not None:
-                tabbed_browser.setCurrentIndex(tab_to_focus)
+                tabbed_browser.currentChanged.emit(tab_to_focus)
```

Comment:
Why would you not actually focus it anymore?

Reply:
I do by emitting the signal, otherwise tabbrowser.on_current_changed() is not called and input_mode is not restored.

Reply:
But why emit the signal but not actually focus the tab anymore?

Reply:
Hmm, this is not working anymore. I tested it earlier and it worked. Ok, back to drawing board...

Reply:
For some reason, when focusing the tab using tabbed_browser.setCurrentIndex(tab_to_focus) it does not restore the input_mode... but if I manually focus another tab and back to this tab, it restores the focus.